### ISSUE TITLE
Ladders cause gravity damage to player

### ISSUE THREAD

**Issue by Heptagon-ru** [anegostudios/VintageStory-Issues#66]
1.5.2.6 client, 1.5.2.7 server i guess, official server.

I place one ladder pole and another one right behind it, ajanced. The second ladder pole is taller. Climb first to the top of the shorter one, then move to the taller one from the top of the shorter one and continue climbing the taller one. Start taking significant fall damage climbing the taller one either along the way or near the top.

First reported in Discord by Milo, search "Standing on top of the shorter ladder I take fall damage." in bugs channel.

Btw, if cut those ladders by axe, first the short ladder pole - it falls fully, as if it is alone. When next cutting a lonely standing second one, the tall one - only the destroyed ladder block is dropped, all the rest ladders are still there, levitating in the air.

I most likely captured it all on my stream.

![2018-04-04_18-56-44](https://user-images.githubusercontent.com/25122345/38334659-9a00d23e-3864-11e8-8374-3f70be71b8ce.png)

![2018-04-04_19-01-27](https://user-images.githubusercontent.com/25122345/38334799-129d924a-3865-11e8-9cb5-20dfaedd0a88.png)

**Comment by copygirl** [comment]
Possible partial duplicate of #64.

**Comment by Heptagon-ru** [comment]
I have just reproduced the issue in singleplayer creative world. 1.5.2.6.
The trick is probably to place the short ladders first and the tall ladders behind them - last.
Also it may be important to place them holding the mouse button, not clicking
On the screenshot you can see that the right pair of ladders doesnt damage me, there I placed the tall ones first.
But the left pair of ladders killed me exactly as described in the ticket, you can see my dropped items there. Also!!! The furthest ladder there was as tall as in the testcase to the right, but after respawning in the world I see it is just one block high. I gues this is the cause of the damage.
![2018-04-07_12-23-47](https://user-images.githubusercontent.com/25122345/38453468-1a354380-3a5f-11e8-8c5e-c3bad1d8abc9.png)

**Comment by sirStroam** [comment]
![image](https://user-images.githubusercontent.com/12073699/38460156-766939fe-3a69-11e8-93cc-cf15f3e54bb8.png)
Might be something

**Comment by copygirl** [comment]
Like @sirStroam mentioned in the [other issue](https://github.com/anegostudios/VintageStory-Issues/issues/64#issuecomment-379494323), I also saw some ladders that were placed onto another disappear after relogging. The out of nowhere fall damage is very likely related to fake client-side-only ladders.

**Comment by sirStroam** [comment]
I think I fixed it in my mod. The ladders don't disappear anymore when relogging at least.

**Comment by Heptagon-ru** [comment]
Still persists in 1.6 rc1, PTS, because not all ladders I place actually are registered.
Seconf screenshot is after relog, with unregistered ladders disappeared.
![2018-08-01_20-51-26](https://user-images.githubusercontent.com/25122345/43539542-d2cab712-95cd-11e8-9acc-7593d4f2af30.png)
![2018-08-01_20-51-53](https://user-images.githubusercontent.com/25122345/43539543-d2eb1192-95cd-11e8-8534-b8579b18626d.png)

**Comment by tyronx** [comment]
Hm ok, but thats a different bug then, a block desync between client and server.

**Comment by copygirl** [comment]
@tyronx I think the block desync is what's being described here in this issue. As seen in the screenshot in the description, the ladders disappear after relogging.

**Comment by copygirl** [comment]
Could it still be that the client tries to check placement requirements for a different rotation than the server? Additionally, if the server refuses to place a block, it should re-send the actual block to the client. Ladders have a little bit of their own placement code, right? It might have to do this manually.

**Comment by Heptagon-ru** [comment]
@tyronx The bug is "sometimes getting gravity damage on top of high ladder". This is how a player experiences it.
The core cause of the bug seems to be some client-server desync, resulting in "ghost" placed ladders.
I guess my investigations are guiding to this conclusion.
Also the bug is not easily reproducible with a single ladder. yesterday in rc1 I caught it only once, reporting right away. In other cases in rc1 the bug does not appear, idk why.
Maybe my previous testcase with two adjancent ladders still works and allows to reproduce the bug more frequently.

**Comment by tyronx** [comment]
Oh ok, true, the bug is the same. Ok well, do let me know of any further cases where it happens

**Comment by tyronx** [comment]
Pretty sure this is fixed by now
